### ISSUE TITLE
Notice: Automattic\Jetpack\Sync\Functions was called incorrectly.

### ISSUE THREAD

**Issue by bobbingwide** [Automattic/jetpack#13048]
This message is being issued every now and then in a local exact clone of the live site.

Notice: Automattic\Jetpack\Sync\Functions was called incorrectly. Not all plugins have loaded yet but we requested the class Automattic\Jetpack\Sync\Functions Please see Debugging in WordPress for more information. (This message was added in version dev-add/jetpack-compat-package.) in C:\apache\htdocs\srab\wp-includes\functions.php on line 4773

#### Steps to reproduce the issue

1. Clone a live site to a local instance.
2. Confirm that it's an exact copy
3. Keep using it with Jetpack activated and WP_DEBUG true
4. See error every now and then

#### What I expected

No Notices

#### What happened instead
The message was produced every now and then.
Stack trace for `doing_it_wrong_run`.
```
0. bw_lazy_backtrace C:\apache\htdocs\wordpress\wp-content\plugins\oik-bwtrace\libs\bwtrace.php:108 0
1. bw_backtrace C:\apache\htdocs\wordpress\wp-content\plugins\oik-bwtrace\includes\bwtrace-actions.php:191 0
2. bw_trace_doing_it_wrong_run(Automattic\Jetpack\Sync\Functions,Not all plugins have loaded yet but we requested the class Automattic\Jetpack\Sync\Functions,dev-add/jetpack-compat-package) C:\apache\htdocs\srab\wp-includes\class-wp-hook.php:286 3
3. apply_filters(,array) C:\apache\htdocs\srab\wp-includes\class-wp-hook.php:310 2
4. do_action(array) C:\apache\htdocs\srab\wp-includes\plugin.php:465 1
5. do_action(doing_it_wrong_run,Automattic\Jetpack\Sync\Functions,Not all plugins have loaded yet but we requested the class Automattic\Jetpack\Sync\Functions,dev-add/jetpack-compat-package) C:\apache\htdocs\srab\wp-includes\functions.php:4746 4
6. _doing_it_wrong(Automattic\Jetpack\Sync\Functions,Not all plugins have loaded yet but we requested the class Automattic\Jetpack\Sync\Functions,dev-add/jetpack-compat-package) C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\vendor\autoload_packages.php:102 3
7. Automattic\Jetpack\Autoloader\autoloader(Automattic\Jetpack\Sync\Functions) C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\vendor\autoload_packages.php:0 1
8. spl_autoload_call(Automattic\Jetpack\Sync\Functions) C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\class.jetpack.php:6192 1
9. get_sync_error_idc_option C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\class.jetpack.php:6133 0
10. validate_sync_error_idc_option C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\class.jetpack.php:6027 0
11. check_identity_crisis C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\class.jetpack-idc.php:53 0
12. __construct C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\class.jetpack-idc.php:44 0
13. init C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\class.jetpack-idc.php:703 0
14. require_once(C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\class.jetpack-idc.php) C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\load-jetpack.php:61 1
15. require_once(C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\load-jetpack.php) C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\jetpack.php:191 1
16. include_once(C:\apache\htdocs\wordpress\wp-content\plugins\jetpack\jetpack.php) C:\apache\htdocs\srab\wp-settings.php:362 1
17. require_once(C:\apache\htdocs\srab\wp-settings.php) C:\apache\htdocs\srab\wp-config.php:92 1
18. require_once(C:\apache\htdocs\srab\wp-config.php) C:\apache\htdocs\srab\wp-load.php:37 1
19. require_once(C:\apache\htdocs\srab\wp-load.php) C:\apache\htdocs\srab\wp-blog-header.php:13 1
20. require(C:\apache\htdocs\srab\wp-blog-header.php) C:\apache\htdocs\srab\index.php:17 1
```
**Screenshots**

### Workaround

Choice | Reason for not doing so
------ | ---------------------
Set `WP_DEBUG` false | This is a development environment. I want to see messages.
Use a local site name without a dot | Some other routines fail when the domain doesn't contain a period
Set `JETPACK_DEV_DEBUG` true | Well, it works but see Note 1.
Implement `jetpack_development_mode` filter | In this instance the filter would work if implemented by a plugin; not a theme. See Note 2.

#### Note 1.
I believe that the logic to check for Identity Crisis is supposed to work regardless of the value of  `JETPACK_DEV_DEBUG`.  In this case there appears to be a little timing issue. 
The problem resolves itself every now and then.  
I don't want to set it to true since there are 57 other checks that would now produce different results.  

#### Note 2.
Example of a filter function that doesn't work in a Genesis child theme since it's not been added by the time the filter is first invoked.
```
add_filter( 'jetpack_development_mode', 'genesis_srab_jetpack_development_mode');
function genesis_srab_jetpack_development_mode( $development_mode ) {
	
	if ( false === $development_mode ) {
		$development_mode = defined( WP_DEBUG ) && WP_DEBUG;
	}
	return $development_mode;
}
```

**Comment by jeherve** [comment]
Could you give it another try today, with Jetpack's development version or the latest Beta available here?
https://downloads.wordpress.org/plugin/jetpack.7.5.3-beta.zip

Let us know if that helps.

**Comment by bobbingwide** [comment]
Hi Jeremy. 7.5.3-beta doesn't fix the problem. The stack trace is basically the same except for 
the line number on this one. 

15. require_once(C:\apache\htdocs\srab\wp-content\plugins\jetpack\load-jetpack.php) C:\apache\htdocs\srab\wp-content\plugins\jetpack\jetpack.php:194 1

PS. I installed the plugin directly, rather than via symlinking, as in the previous stack trace

**Comment by suzannealdrich** [comment]
I'm also still getting this error in my dev environment after installing 7.5.3 tonight.

**Comment by jeherve** [comment]
Somewhat related: #3827

**Comment by chrisjimallen** [comment]
Thats a +1 for me on my staging server using a `staging.` subdomain: 

> Notice: Automattic\Jetpack\Sync\Functions was called incorrectly. Not all plugins have loaded yet but we requested the class Automattic\Jetpack\Sync\Functions Please see Debugging in WordPress for more information. (This message was added in version dev-add/jetpack-compat-package.) in /var/www/html/development/mysite/wp-includes/functions.php on line 4773

**Comment by jeherve** [comment]
@chrisjimallen I can't seem to reproduce.

- Do you use Jetpack 7.5.3?
- Do you use any other plugins on that site? 
- When you go to the Jetpack dashboard, do you see the following notice?
![image](https://user-images.githubusercontent.com/426388/61526409-a83dba00-aa1a-11e9-9bf1-11a3ef4c799e.png)

Thank you.

**Comment by chrisjimallen** [comment]
- Yes, I'm running 7.5.3. Oddly, the warning seems to have gone away when I checked it today :-/
- There are other plugins, quite a lot.
- This is what I see now:

<img width="1525" alt="Screenshot 2019-07-22 at 11 50 11" src="https://user-images.githubusercontent.com/1681063/61627158-fcdd7100-ac76-11e9-92f0-f359ecdd93e6.png">

I'm going to say that this is no longer an issue for me personally.

**Comment by rebeccahum** [comment]
I'm experiencing a similar notice in https://github.com/rebeccahum/VIP-Jetpack-Sync-Cron-Only/issues/2 when the Jetpack functions are being called too early in my drop-in plugin.  A workaround I found was to hook onto the initialization at a later hook after JP has been fully loaded.

**Comment by rogercoathup** [comment]
Also seeing this error on a staging site

**Comment by simplenotezy** [comment]
Same issue here, on local development

**Comment by chrisjimallen** [comment]
> Same issue here, on local development

Glad its not just me

**Comment by abhishek-pokhriyal** [comment]
Hey, 

Facing this on my staging. Is it something to do we the plugin?

**Comment by jeherve** [comment]
> Is this something to do with the plugin?

We haven't been able to figure this out yet, but we'll update this issue when we do!

**Comment by keepsmilyn** [comment]
Yes, also seeing this issue in our logs and when turning on debugging to display in HTML. 

`Error message: "Automattic\Jetpack\Sync\Functions was called <strong>incorrectly</strong>. Not all plugins have loaded yet but we requested the class Automattic\Jetpack\Sync\Functions Please see <a href="https://codex.wordpress.org/Debugging_in_WordPress">Debugging in WordPress</a> for more information. (This message was added in version 9999999-dev.) in /......../wp-includes/functions.php on line 4778"`

**Comment by kojaa** [comment]
So this happened to me as well, the only fix was just to delete the plugin and never install it again since it fucked up last 2 times on updates. 
At first i couldn't even delete it through plugin interface but deleting the folder fixed issues :)

**Comment by kraftbj** [comment]
@kojaa If even deleting the plugin through the plugin interface is failing, that's most common due to something like incorrect server permissions or a server that is failing during the file operation. It wouldn't be a Jetpack issue.

**Comment by pfernandez** [comment]
Adding `define( 'JETPACK_DEV_DEBUG', true );` to wp-config.php has silenced the notice for me.

**Comment by kbrown9** [comment]
Thanks for the reports and stack traces!

I see two paths to this bug in these reports. One path includes a call to `Jetpack_IDC::init()`, and that should be fixed by PR #14030. In that PR, `Jetpack_IDC::init()` is hooked to plugins_loaded so the PHP Notice should not be generated.

The second path goes through `Jetpack_Admin::init()`, and it will need a separate fix.
